alf/experience_replayers/replay_buffer.py

```diff
@@ -134,6 +144,63 @@ def add_batch(self, batch, env_ids=None):
             self._current_size[env_ids] = torch.min(
                 current_size + 1, torch.tensor(self._max_length))
 
+    def dequeue(self, env_ids=None):
+        """Return earliest batch and remove it from buffer."""
+        with alf.device(self._device):
+            env_ids = self.check_convert_env_ids(env_ids)
+            assert len(env_ids.shape) == 1, "env_ids should be an 1D tensor"
+
+            current_size = self._current_size[env_ids]
+            min_size = current_size.min()
+            assert min_size >= 1, (
+                "Not all environments have enough data. The smallest data "
+                "size is: %s Try storing more data before calling dequeue" %
+                min_size)
+            pos = self._current_pos[env_ids] - current_size
+            pos = pos.reshape(-1, 1)  # [B, 1]
+            pos = pos % self._max_length
+            env_ids = env_ids.reshape(-1, 1).repeat(1, 1)  # [B, T]
```

Comment:
where it T from? Is T equal to 1? Since there is only one step, we should not have T.

Reply:
Good point.